src/views/main-view.js

```diff
@@ -149,6 +130,36 @@ MainView.prototype._initialize = function () {
   this._sendToDefaultView();
 };
 
+MainView.prototype._onChangeActivePaymentMethodView = function (id) {
+  var activePaymentView = this.getView(id);
+
+  if (id === PaymentMethodsView.ID) {
+    classList.add(this.paymentMethodsViews.container, 'braintree-methods--active');
+    classList.remove(this.sheetContainer, 'braintree-sheet--active');
+    this.vaultedCardSelectAnalyticEvent();
```

Comment:
I think where you actually want this logic is in here: https://github.com/braintree/braintree-web-drop-in/blob/c8bc81b64def5ef739ea5b42db0c65c7d1fc75ce/src/views/payment-method-view.js#L92-L97

You can check if the selected method is a vaulted payment method by checking `this.paymentMethod.vaulted` in that view. So the method would look something like:

```js
PaymentMethodView.prototype._choosePaymentMethod = function () {
  if (this.model.isInEditMode()) {
    return;
  }
  if (this.paymentMethod.vaulted) {
    analytics.sendEvent(this.client, 'vaulted-' + this.paymentMethod.type + '.select');
  }

  this.model.changeActivePaymentMethod(this.paymentMethod);
};
```

You'll need https://github.com/braintree/braintree-web-drop-in/commit/5ff489d0cd92bdde28d95cff1b75e597aa0ef07d so that the client is passed along to the payment method view.

Reply:
This is a much better solution. Thanks for the input and thought![ Revisions here.](https://github.com/braintree/braintree-web-drop-in/pull/508/commits/10a8aaae4e79e37e3159e8680dbb129f7df2a1fe)